api/db/activity/activity.js

```diff
@@ -0,0 +1,77 @@
+const logger = require('../../logger')('db activity model');
+
+module.exports = {
+  apdActivity: {
+    tableName: 'activities',
+
+    apd() {
+      return this.belongsTo('apd');
+    },
+
+    goals() {
+      return this.hasMany('apdActivityGoal');
+    },
+
+    approaches() {
+      return this.hasMany('apdActivityApproach');
+    },
+
+    expenses() {
+      return this.hasMany('apdActivityExpense');
+    },
+
+    schedule() {
+      return this.hasMany('apdActivitySchedule');
+    },
+
+    static: {
+      updateableFields: ['name', 'description'],
+      owns: {
+        goals: 'apdActivityGoal',
+        approaches: 'apdActivityApproach',
+        expenses: 'apdActivityExpense',
+        schedule: 'apdActivitySchedule'
+      },
+      foreignKey: 'activity_id',
+      withRelated: [
+        'approaches',
+        'goals',
+        'goals.objectives',
+        'expenses',
+        'expenses.entries',
+        'schedule'
+      ]
+    },
```

Comment:
I want to revisit this pattern in the future because there's "secret knowledge" going on.  That is, it's extending bookshelf in a way that's not documented and could confuse a dev new to the project.

I definitely prefer having this little bit of metadata attached to the model over having gobs of very similar code on every model, though.